codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -335,18 +335,20 @@ update these file paths accordingly.
 
       log.info('uploading ${args.join(' ')}');
       final io.ProcessResult result = processManager.runSync(args);
+      if (result.exitCode < 0 || result.exitCode > 255) {
+        throw CodesignException(
+            'Command "${args.join(' ')}" failed with exit code ${result.exitCode}\nStdout: ${result.stdout}\nStderr: ${result.stderr}');
+      }
 
       final String combinedOutput = (result.stdout as String) + (result.stderr as String);
       final RegExpMatch? match;
       match = _notarytoolRequestPattern.firstMatch(combinedOutput);
 
       if (match == null) {
-        if (result.exitCode >= 0 && result.exitCode <= 255) {
-          log.warning('The upload to notary service completed normally, but'
-              ' the output format does not match the current notary tool version.'
-              ' If after inspecting the output below, you believe the process finished '
-              'successfully but was not detected, please contact fujino@');
-        }
+        log.warning('The upload to notary service completed normally, but'
```

Comment:
I think we want to fail the build here. I've learned from experience, if the build succeeds, nobody reads the logs.

Reply:
Thanks for the suggestion! I moved this warning to the place where it is after using up all retry counts

Reply:
umm I think if the output does not match because of notary tool version, it will fail consistently and eventually throw an exception. but yeah i moved it to right before we throw the exception so it is more obvious.